File: android/model/Fact.kt
package com.tobiasheymann.ep.model

import java.io.Serializable

data class Fact(
    val icon: FactIconType,
    val text: String
) : Serializable {
    companion object {
        fun toObject(map: Map<String, Any>): Fact? {
            return if (
                map["icon"] == null
                || map["text"] == null
            )
                null
            else
                Fact(
                    FactIconType(map["icon"]!! as String)!!,
                    map["text"]!! as String
                )
        }
    }
}

File: ios/model/Fact.swift
struct Fact: Serializable {
    let icon: FactIconType
    let text: String
    static func toObject(map: [String: Any]) -> Fact? {
        return map["icon"] == nil || map["text"] == nil ? nil : Fact(icon: FactIconType(rawValue: map["icon"]! as! String)!, text: map["text"]! as! String)
    }
}
